src/arduino/arduino.ts

```diff
@@ -213,7 +213,7 @@ export class ArduinoApp {
         UsbDetector.getInstance().pauseListening();
         await vscode.workspace.saveAll(false);
 
-        const appPath = path.join(ArduinoWorkspace.rootPath, dc.sketch);
+        const appPath = this.isArduinoCli() ? ArduinoWorkspace.rootPath : path.join(ArduinoWorkspace.rootPath, dc.sketch);
         const args = ["--upload", "--board", boardDescriptor, "--port", dc.port, "--useprogrammer",

```

Comment:
Upload for the cli should probably use 'arduino-cli compile --upload ...'

Reply:
I refactored this method, upload using programmer is now an option within the upload method.